JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -110,60 +119,68 @@ fun HomeRoute(
+                    openDrawer = openDrawer,
+                    homeListLazyListState = homeListLazyListState,
+                    articleDetailLazyListStates = articleDetailLazyListStates,
+                    snackbarHostState = snackbarHostState,
+                    onSearchInputChanged = onSearchInputChanged,
+                )
+            }
+            HomeScreenType.Feed -> {
+                HomeFeedScreen(
+                    uiState = uiState,
+                    showTopAppBar = !isExpandedScreen,
+                    onToggleFavorite = onToggleFavorite,
+                    onSelectPost = onSelectPost,
+                    onRefreshPosts = onRefreshPosts,
+                    onErrorDismiss = onErrorDismiss,
+                    openDrawer = openDrawer,
+                    homeListLazyListState = homeListLazyListState,
+                    snackbarHostState = snackbarHostState,
+                    onSearchInputChanged = onSearchInputChanged,
+                )
+            }
+            HomeScreenType.ArticleDetails -> {
+                // Guaranteed by above condition for home screen type
+                check(uiState is HomeUiState.HasPosts)
 
-            ArticleScreen(
-                post = uiState.selectedPost,
-                isExpandedScreen = isExpandedScreen,
-                onBack = onInteractWithFeed,
-                isFavorite = uiState.favorites.contains(uiState.selectedPost.id),
-                onToggleFavorite = {
-                    onToggleFavorite(uiState.selectedPost.id)
-                },
-                lazyListState = articleDetailLazyListStates.getValue(
-                    uiState.selectedPost.id
+                ArticleScreen(
+                    post = uiState.selectedPost,
+                    isExpandedScreen = isExpandedScreen,
+                    onBack = onInteractWithFeed,
+                    isFavorite = uiState.favorites.contains(uiState.selectedPost.id),
+                    onToggleFavorite = {
+                        onToggleFavorite(uiState.selectedPost.id)
+                    },
+                    lazyListState = articleDetailLazyListStates.getValue(
+                        uiState.selectedPost.id
+                    )
                 )
-            )
 
-            // If we are just showing the detail, have a back press switch to the list.
-            // This doesn't take anything more than notifying that we "interacted with the list"
-            // since that is what drives the display of the feed
-            BackHandler {
-                onInteractWithFeed()
+                // If we are just showing the detail, have a back press switch to the list.
+                // This doesn't take anything more than notifying that we "interacted with the list"
+                // since that is what drives the display of the feed
+                BackHandler {
+                    onInteractWithFeed()
+                }
```

Comment:
I'm wondering if the `BackHandler` should be outside of the `AnimatedContent`, so that it is introduced and removed immediately without being part of the animation.

Otherwise if the article details is animating away due to going back, I think this `BackHandler` will still be present during that animation.

Reply:
Good point, I'll move it.